docs/dev_cycle.part-03.rst

```diff
@@ -0,0 +1,194 @@
+on the mainnet.
+
+.. note::
+  To deploy a *new* ``Donator`` contract, maybe to another blockchain, or to another address on the same blockchain,
+  we also need a Web3 contract *object*, but this time the object does require a compiled
+  "contract"
+
+Actually when we say ""contract", we deal with three different things:
+
+#. **Contract Instance**: Bytecode on a blockchain
+#. **Solidiy Source**: A Solidity contract definition, like ``contract Donator {...}``
+#. **Web3 Object**: An object that provides methods to deploy a new contract instance, or interact with an existing one
+
+
+Testing a Contract
+------------------
+
+Get the contract object
+'''''''''''''''''''''''
+
+The answer to the question "what are we actually testing" is: We test a bytecode program that runs on a blockchain. We test
+a contract *instance*.
+
+Ideally, for testing, we would need to take the Solidity source file, compile it, deploy it to a blockchain, create a Web3 contract
+object that points to this instance, and handover this object to the test function so we can test it.
+
+And here is where you will start to appreciate Populus, **which does all that for you in one line of code.**
+
+Add the first test:
+
+.. code-block:: shell
+
+  $ nano tests/test_donatory.py
+  
+The test file should look as follows:
+
+.. code-block:: python
+
+  def test_default_usd_rate(chain):
+      donator, deploy_tx_hash _ = chain.provider.get_or_deploy_contract('Donator')
+      default_usd_rate = donator.call().default_usd_rate()
+      assert default_usd_rate == 350
+      
+      
+The magic happens with ``get_or_deploy_contract``. This function gets an existing contract if it exists on the blockchain, and if it
+doesn't, it compiles the Solidity source, deploys it to the blockchain, creates a ``Contract`` object, exposes the deployed contract
+as a *python object with python functions*, and returns this object to the test function.
+
+From this point onward, you have a *Python* object, with *Python* methods, that correspond to the original deployed contract
+bytecode on the blockchain. Cool, isn't it?
+
+.. note::
+  
+    For the contract name you use the Solidity contract name, ``Donator``, and *not* the file name, ``Donator.sol``.
+    A Solidity source file can include more than one contract definition (as a Python file can include more than one class definition).
+
+Get the blockchain
+''''''''''''''''''
+
+Another bonus is the ``chain``, at ``def test_default_usd_rate(chain)``. It gives the test function a Python object 
+that corresponds to a running blockchain, the ``tester`` blockchain.
+Reminder: The ``tester`` chain is ephemeral, saved only in memory, and will reset on every test run.
+
+The ``chain`` argument is a py.test *fixture*: in py.test world it's a special argument that the test function can accept.
+You don't have to declare or assign it, it's just ready and available for your function.
```

Comment:
Maybe change to:

> `py.test` automatically *injects* these arguments into your test functions for you.

And maybe add a link to the pytest documentation on fixtures for *further reading*